crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/cheat_execution_info.rs

```diff
@@ -0,0 +1,190 @@
+    pub block_timestamp: Operation<u64>,
+    pub sequencer_address: Operation<ContractAddress>,
+}
+
+#[derive(FromReader, Clone, Default, Debug)]
+pub struct ExecutionInfoMockOperations {
+    pub block_info: BlockInfoMockOperations,
+    pub tx_info: TxInfoMockOperations,
+    pub caller_address: Operation<ContractAddress>,
+    pub contract_address: Operation<ContractAddress>,
+    pub entry_point_selector: Operation<EntryPointSelector>,
+}
+
+macro_rules! for_all_fields {
+    ($macro:ident!) => {
+        $macro!(caller_address);
+        $macro!(contract_address);
+        $macro!(entry_point_selector);
+
+        $macro!(block_info.block_number);
+        $macro!(block_info.block_timestamp);
+        $macro!(block_info.sequencer_address);
+
+        $macro!(tx_info.version);
+        $macro!(tx_info.account_contract_address);
+        $macro!(tx_info.max_fee);
+        $macro!(tx_info.signature);
+        $macro!(tx_info.transaction_hash);
+        $macro!(tx_info.chain_id);
+        $macro!(tx_info.nonce);
+        $macro!(tx_info.resource_bounds);
+        $macro!(tx_info.tip);
+        $macro!(tx_info.paymaster_data);
+        $macro!(tx_info.nonce_data_availability_mode);
+        $macro!(tx_info.fee_data_availability_mode);
+        $macro!(tx_info.account_deployment_data);
+    };
+}
+
+impl CheatnetState {
+    fn get_cheated_execution_info_for_contract(
+        &mut self,
+        target: ContractAddress,
+    ) -> &mut ExecutionInfoMock {
+        self.cheated_execution_info_contracts
+            .entry(target)
+            .or_insert_with(|| self.global_cheated_execution_info.clone())
+    }
+
+    pub fn cheat_execution_info(&mut self, execution_info_mock: ExecutionInfoMockOperations) {
+        macro_rules! cheat {
+            ($($path:ident).+) => {
+                match execution_info_mock.$($path).+ {
+                    Operation::Retain => {}
+                    Operation::Start(CheatArguments {
+                        value,
+                        span,
+                        target,
+                    }) => {
+                        let cheated_info = self.get_cheated_execution_info_for_contract(target);
+
+                        cheated_info.$($path).+ = CheatStatus::Cheated(value, span);
+                    }
+                    Operation::Stop(target) => {
+                        let cheated_info = self.get_cheated_execution_info_for_contract(target);
+
+                        cheated_info.$($path).+ = CheatStatus::Uncheated;
+                    }
+                    Operation::StartGlobal(value) => {
+                        self.global_cheated_execution_info.$($path).+ =
+                            CheatStatus::Cheated(value.clone(), CheatSpan::Indefinite);
+
+                        for val in self.cheated_execution_info_contracts.values_mut() {
+                            val.$($path).+ = CheatStatus::Cheated(value.clone(), CheatSpan::Indefinite);
+                        }
```

Comment:
That's kinda expensive, but I guess the logic is simpler

Reply:
Actually, both `.clone()` and `HashMap` iteration are not expensive here. The cloned value is the value of one of the fields listed in the `for_all_fields!` macro (`Vec` in the worst case, `Felt252` commonly). `self.cheated_execution_info_contracts` will contain only contracts that have previously been cheated. There should not be a lot of them, as this is per test.